src/main/java/uk/co/ramp/covid/simulation/population/Population.java

```diff
@@ -281,13 +289,52 @@ public void timeStep(Time t, DailyStats dStats) {
         households.forEach(h -> h.doTesting(t));
 
         // Movement places people in "next" buffers (to avoid people moving twice in an hour)
-        households.forEach(h -> h.doMovement(t, lockdown));
-        places.getAllPlaces().forEach(p -> p.doMovement(t, lockdown));
-
-        households.forEach(h -> h.stepPeople());
-        places.getAllPlaces().forEach(p -> p.stepPeople());
-    }
-
+        households.forEach(h -> h.decideOnMovement(t, lockdown));
+        places.getAllPlaces().forEach(p -> p.decideOnMovement(t, lockdown));
+
+        // do infections while travelling
+        doAllTravelInfect(t, dStats);
+
+        // apply movements
+        households.forEach(h -> h.implementMovement());
+        places.getAllPlaces().forEach(p -> p.implementMovement());
+    }
+
+	public void doAllTravelInfect(Time t, DailyStats dStats) {
+		// look for similar journeys - currently simple but could be made more complex considering travel types and other aspects
+		
+		for (Place place : places.getAllPlaces()) {
+			// People going to the same place
+			doTravelInfect(place.getPeopleMovingToHere(), t, dStats);
+			
+			// People going from the same place
+			doTravelInfect(place.getPeopleMovingFromHere(), t, dStats);
+		}
+	}
+
+	// this method could might not be best place in the class
+	public void doTravelInfect(Collection<Person> peopleOnSimilarJourneys, Time t, DailyStats dStats) {
```

Comment:
Just a note for later this will need to take the form of the other "doInfects" (now we've fixed the bugs).